### PR TITLE
feat(noResults): allow providing URL to report missing results

### PR DESCRIPTION
**Pull request by shortcuts**
## Summary

This PR provides an option for the user to define a link to report the missing results, nothing is rendered by default.

With the deprecation of our legacy infra (follow up of https://github.com/algolia/docsearch-configs/pull/5069), the current value of this link (https://github.com/algolia/docsearch-configs) does not make sense anymore.

## Other solutions

- Remove this informative message
  - We did not went with this as it is a widely used feature, see: https://github.com/algolia/docsearch-configs/issues
- Allow providing a function or JSX element in order to retrieve current query
  - We did not receive feature requests to customize the `noResults` screen, which is also doable by importing the DocSearch components.

### CODE REVIEW

File: packages/docsearch-react/src/DocSearchModal.tsx [hunk 1]
```diff
@@ -50,6 +50,7 @@ export function DocSearchModal({
   disableUserPersonalization = false,
   initialQuery: initialQueryFromProp = '',
   translations = {},
+  reportMissingResultsUrl,
```
Comment (bodinsamuel) [inline packages/docsearch-react/src/DocSearchModal.tsx:53]: it's optional but there is are no default?
edit: the component is optional okay 🆗

**Review by bodinsamuel — APPROVED** [review]

File: packages/docsearch-react/src/NoResultsScreen.tsx [hunk 2]
```diff
@@ -68,17 +68,18 @@ export function NoResultsScreen({
...
-        >
-          {openIssueLinkText}
-        </a>
-        .
-      </p>
+      {props.reportMissingResultsUrl && (
+        <p className="DocSearch-Help">
+          {`${reportMissingResultsText} `}
+          <a
+            href={props.reportMissingResultsUrl}
```
Comment (francoischalifour) [inline packages/docsearch-react/src/NoResultsScreen.tsx:75]: Why don't we pass the query to a function? It [seems to be widely used](https://github.com/algolia/docsearch-configs/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+missing+results) and makes the user request accurate. We're loosing information with this update.

Comment (shortcuts) [reply packages/docsearch-react/src/NoResultsScreen.tsx:75]: Do you mean what is suggested in the `Other solutions`?

We could definitely have `reportMissingResultsUrl` as a function `(query:string) => string` so the user can build the appropriate link.

Comment (francoischalifour) [reply packages/docsearch-react/src/NoResultsScreen.tsx:75]: `({ query }) => string`, yes.

Comment (shortcuts) [reply packages/docsearch-react/src/NoResultsScreen.tsx]: works as expected: https://github.com/algolia/docsearch/pull/1289/commits/09346644629173af0287c8b4f1b7c15254e1bb0d#diff-f2decf6ff1d6fc42353fede839f1280094117430c1253a0359a54ea725d36a3aR256-R258

Comment (shortcuts) [reply packages/docsearch-react/src/NoResultsScreen.tsx]: Not sure about the name 🤔

File: packages/website/docs/api.mdx [hunk 3]
```diff
@@ -170,6 +171,23 @@ const translations: DocSearchTranslations = {
 </div>
 </details>
 
+## `reportMissingResultsUrl`
+
+> `type: ({ query:string }) => string` | **optional**
```
Comment (francoischalifour) [inline packages/website/docs/api.mdx:176]:
```suggestion
> `type: ({ query: string }) => string` | **optional**
```

File: packages/website/docs/api.mdx [hunk 4]
```diff
@@ -309,6 +327,23 @@ const translations: DocSearchTranslations = {
...
```
Comment (francoischalifour) [inline packages/website/docs/api.mdx:332]:
```suggestion
> `type: ({ query: string }) => string` | **optional**
```

File: packages/docsearch-react/src/NoResultsScreen.tsx [hunk 5]
```diff
@@ -68,17 +68,18 @@ export function NoResultsScreen({
...
-        >
-          {openIssueLinkText}
-        </a>
-        .
-      </p>
+      {props.reportMissingResultsUrl && (
+        <p className="DocSearch-Help">
+          {`${reportMissingResultsText} `}
+          <a
+            href={props.reportMissingResultsUrl({ query: props.state.query })}
```
Comment (francoischalifour) [inline packages/docsearch-react/src/NoResultsScreen.tsx:75]: `getMissingResultsUrl()` may make more sense now?

Comment (shortcuts) [reply packages/docsearch-react/src/NoResultsScreen.tsx]: Yep!

**Review by francoischalifour — APPROVED** [review]

**Pull request merged**